runtime/metricsview/executor.go

```diff
@@ -328,3 +332,288 @@ func (e *Executor) Export(ctx context.Context, qry *Query, executionTime *time.T
+		finalArgs []any
+		rowsCap   int64
+		err       error
+	)
+	for i, d := range qry.Dimensions {
+		if i > 0 {
+			finalSQL.WriteString(" UNION ALL ")
+		}
+		q := &Query{
+			MetricsView:         qry.MetricsView,
+			Dimensions:          []Dimension{{Name: d}},
+			Measures:            nil,
+			PivotOn:             nil,
+			Spine:               nil,
+			Sort:                nil,
+			TimeRange:           qry.TimeRange,
+			ComparisonTimeRange: nil,
+			Where:               nil,
+			Having:              qry.Having,
+			Limit:               qry.Limit,
+			Offset:              nil,
+			TimeZone:            "",
+			Label:               false,
+		}
+		q.Where = whereExprForSearch(qry.Where, d, qry.Search)
+
+		if err := e.rewriteQueryTimeRanges(ctx, q, executionTime); err != nil {
+			return nil, err
+		}
+
+		rowsCap, err = e.rewriteQueryEnforceCaps(q)
+		if err != nil {
+			return nil, err
+		}
+
+		ast, err := NewAST(e.metricsView, e.security, q, e.olap.Dialect())
+		if err != nil {
+			return nil, err
+		}
+
+		if err := e.rewriteLimitsIntoSubqueries(ast); err != nil {
+			return nil, err
+		}
+
+		sql, args, err := ast.SQL()
+		if err != nil {
+			return nil, err
+		}
+		finalSQL.WriteString(fmt.Sprintf("SELECT %s AS dimension, %s AS value FROM (%s)", e.olap.Dialect().EscapeStringValue(d), e.olap.Dialect().EscapeIdentifier(d), sql))
+		finalArgs = append(finalArgs, args...)
+	}
+
+	res, err := e.olap.Execute(ctx, &drivers.Statement{
+		Query:            finalSQL.String(),
+		Args:             finalArgs,
+		Priority:         e.priority,
+		ExecutionTimeout: defaultInteractiveTimeout,
+	})
+	if err != nil {
+		return nil, err
+	}
+	defer res.Close()
+	if rowsCap > 0 {
+		res.SetCap(rowsCap)
+	}
+	searchResult := make([]SearchResult, 0)
+	for res.Next() {
+		var row SearchResult
+		if err := res.Scan(&row.Dimension, &row.Value); err != nil {
+			return nil, err
+		}
+		searchResult = append(searchResult, row)
+	}
+	if res.Err() != nil {
+		return nil, res.Err()
+	}
+	return searchResult, nil
+}
+
+func (e *Executor) executeSearchInDruid(ctx context.Context, olap drivers.OLAPStore, a *AST, search string) ([]SearchResult, error) {
+	if a.Root.FromSelect != nil {
+		// This means either the dimension uses an unnest or measure filters which are not directly supported by native search.
+		// This can be supported in future using query datasource in future if performance turns out to be a concern.
+		return nil, errDruidNativeSearchUnimplemented
+	}
+	var query map[string]interface{}
+	if a.Root.Where != nil {
+		// NOTE :: this does not work for measure filters.
+		// The query planner resolves them to joins instead of filters.
+		rows, err := olap.Execute(ctx, &drivers.Statement{
+			Query:            fmt.Sprintf("EXPLAIN PLAN FOR SELECT 1 FROM %s WHERE %s", *a.Root.FromTable, a.Root.Where.Expr),
+			Args:             a.Root.Where.Args,
+			DryRun:           false,
+			Priority:         0,
+			LongRunning:      false,
+			ExecutionTimeout: 0,
```

Comment:
Should these use the same args as above (in the `Search` implementation)? Otherwise, it's fine to omit them – my suggestion around including empty fields in `Query` only applies to the few structs where it's critical to think about all fields (similar to the project insert/update structs); in normal cases it's totally fine to omit empty fields

Reply:
Sure. I just refactored this code and didn't realise that some of these args can be skipped.